src/main/java/de/tum/in/www1/artemis/web/rest/UserJWTController.java

```diff
@@ -101,44 +102,29 @@ public ResponseEntity<JWTToken> authorizeSAML2(@RequestBody final String body) {
         log.debug("SAML2 authentication: {}", authentication);
 
         try {
-            authentication = saml2Service.get().handleAuthentication(principal);
+            saml2Service.get().handleAuthentication(principal);
         }
         catch (UserNotActivatedException e) {
             // If the exception is not caught a 401 is returned.
             // That does not match the actual reason and would trigger authentication in the client
             return ResponseEntity.status(HttpStatus.FORBIDDEN).header("X-artemisApp-error", e.getMessage()).build();
         }
 
-        final String jwt = tokenProvider.createToken(authentication, rememberMe);
-        final HttpHeaders httpHeaders = new HttpHeaders();
-        httpHeaders.add(JWTFilter.AUTHORIZATION_HEADER, "Bearer " + jwt);
-        return new ResponseEntity<>(new JWTToken(jwt), httpHeaders, HttpStatus.OK);
+        final boolean rememberMe = Boolean.parseBoolean(body);
+        ResponseCookie responseCookie = jwtCookieService.buildLoginCookie(rememberMe);
+        response.addHeader(HttpHeaders.SET_COOKIE, responseCookie.toString());
+
+        return ResponseEntity.ok().build();
```

Comment:
here's the problem. we generated the token based on the authentication. now that's not the case